admin/services/group.service.js

```diff
@@ -86,4 +86,18 @@ groupService.create = async (groupName, groupPupils, schoolId) => {
   return newGroup.insertId
 }
 
+/**
+ * Find groups that have pupils that can get PINs assigned.
+ * @param schoolId
+ * @param pupils
+ * @returns {Promise<*>}
+ */
+groupService.filterGroupsByPupil = async (schoolId, pupils) => {
```

Comment:
Should this be called `findGroupsByPupil`?  It doesn't seem to be filtering at all.  And is it taking the whole of a pupil object as input? Is the whole object required or would an array of pupilIds work?

Reply:
Method renamed. 
'sqlFindGroupsByIds' is doing the clean-up needed, I would rather avoid going through this object twice.